astropy/time/core.py

```diff
@@ -1269,13 +1310,13 @@ def __sub__(self, other):
         # we need a constant scale to calculate, which is guaranteed for
         # TimeDelta, but not for Time (which can be UTC)
         if other_is_delta:  # T - Tdelta
+            out = self.replicate()
             if self.scale in other.SCALES:
-                out = self.replicate()
                 if other.scale not in (out.scale, None):
                     other = getattr(other, out.scale)
             else:
-                out = getattr(self, (other.scale if other.scale is not None
-                                     else 'tai'))
+                out = getattr(out, (other.scale if other.scale is not None
```

Comment:
And just to make sure I wasn't going crazy, I reverted to the original version for this code and replicated the original test failure that drove this change:
```
    def test_add_vector(self):
        """Check time arithmetic as well as properly keeping track of whether
            a time is a scalar or a vector"""
        t = Time(0.0, format='mjd', scale='utc')
        t2 = Time([0.0, 1.0], format='mjd', scale='utc')
        dt = TimeDelta(100.0, format='jd')
        dt2 = TimeDelta([100.0, 200.0], format='jd')
    
        out = t + dt
        assert allclose_jd(out.mjd, 100.0)
        assert out.isscalar
    
        out = t + dt2
>       assert allclose_jd(out.mjd, [100.0, 200.0])
E       assert allclose_jd(100.0, [100.0, 200.0])
E        +  where 100.0 = <Time object: scale='utc' format='mjd' value=100.0>.mjd
```